script.js

```diff
@@ -1,12 +1,377 @@
 //You can edit ALL of the code here
-function setup() {
-  const allEpisodes = getAllEpisodes();
+
+// Level 350
+
+let allEpisodes;
+let showID;
+
+function getAllEpisodes(showID) {
+  fetch(`https://api.tvmaze.com/shows/${showID}/episodes`)
+    .then(response => {
+      if (response.ok) {
+        return response.json();
+      }
+      throw new Error(
+        `Encountered something unexpected: ${response.status} ${response.statusText}`
+      );
+    })
+    .then(data => {
+      allEpisodes = data;
+      setUpEpisodesList()
+    })
+    .catch(error => {
+      let errorMessage = document.createElement("p");
+      errorMessage.innerText = error;
+
+      let rootElem = document.getElementById("root");
+      rootElem.appendChild(errorMessage)
+    })
+}
+
+function setUpEpisodesList() {
   makePageForEpisodes(allEpisodes);
+  addSearchBar();
+  addEpisodesSelector();
+  addShowsSelector();
+  goBackToShowsListPage();
 }
 
-function makePageForEpisodes(episodeList) {
+// Level 100
+
+function makePageForEpisodes(episodesList) {
   const rootElem = document.getElementById("root");
-  rootElem.textContent = `Got ${episodeList.length} episode(s)`;
+
+  let ourSection = document.createElement("section");
+  ourSection.classList.add("allEpisodes")
+  rootElem.appendChild(ourSection);
+
+  for (let episodeIndex = 0; episodeIndex < episodesList.length; episodeIndex++) {
+    let episodeContent = document.createElement("article");
+    episodeContent.classList.add("wrapper");
+
+    let episodeTitle = document.createElement("h3");
+    episodeTitle.classList.add("episodeName");
+
+    let imageContainer = document.createElement("div");
+    imageContainer.classList.add("EpisodePictureContainer")
+
+    let episodeImage = document.createElement("img");
+    episodeImage.classList.add("episodePicture")
+    imageContainer.appendChild(episodeImage)
+
+    let episodeSummary = document.createElement("div");
+    episodeSummary.classList.add("episodeDescription")
+
+    episodeContent.append(episodeTitle, imageContainer, episodeSummary);
+    ourSection.append(episodeContent);
+
+    addContentToEachEpisode(episodesList, episodeIndex);
+  }
+}
+
+function addContentToEachEpisode(list, index) {
+  let title = document.getElementsByClassName("episodeName")[index];
+  title.innerHTML = `${list[index].name} - ${getSeasonAndEpisodeNumber(list, index)}`;
+
+  let picture = document.getElementsByClassName("episodePicture")[index];
+  picture.setAttribute("src", `${getImage(list, index)}`)
+
+  let description = document.getElementsByClassName("episodeDescription")[index];
+  description.innerHTML = `${list[index].summary}`;
+}
+
+function getImage(list, index) {
+  if (list[index].image === null) {
+    return "https://upload.wikimedia.org/wikipedia/commons/a/ac/No_image_available.svg";
+  } else {
+    return list[index].image.medium;
+  }
+
+}
+
+function getSeasonAndEpisodeNumber(list, index) {
+  const pattern = 10;
```

Comment:
Hmmm i think the name `pattern` is a bit confusing. How about `doubleDigits`? So, then if I read line 95 it would say "If the season number is less than a double digit number, add a zero beforehand"